include/flatbuffers/buffer.h

```diff
@@ -75,35 +94,59 @@ template<typename T> struct IndirectHelper {
   typedef T return_type;
   typedef T mutable_return_type;
   static const size_t element_stride = sizeof(T);
-  static return_type Read(const uint8_t *p, uoffset_t i) {
+
+  static return_type Read(const uint8_t *p, const size_t i) {
```

Comment:
this is such a frequently used function where speed is the most important.. I wonder if its worth overloading for 32-vs-64.. just in case some compiler somewhere generates worse code if a 32-bit value is cast to `size_t` without it being necessary?

Reply:
Yep, I shouldn't use `size_t` but instead use the type of the offset itself. That way for 32-bit it is `uoffset_t` and for 64-bit it is `uoffset64_t`. This will avoid casts and code duplication.